README.md

```diff
@@ -87,281 +110,201 @@ Genode:
       end if;
-      return 0;
-   end Block_Size;
-   
-   procedure Initialize_Server (S : Block.Server_Instance; L : String)
-   is
-   begin
-      if Block_Server.Get_Instance (Server) = S then
-         Cai.Log.Client.Info ("Initializing server with label " & L);
+      if Componolit.Gneiss.Timer.Initialized (Timer) then
+         Timer_Client.Finalize (Timer);
       end if;
-   end Initialize_Server;
-   
+   end Destruct;
+
 end Component;
 ```
 
-Similar to the client the generic packages are instantiated consecutively.
-The dispatcher is then used like a regular client.
-With the register method it announces its readiness to receive session requests.
-When a clients want to open a session the `Dispatch` procedure is called.
-It gets the session information and checks if a valid request is available and if it has a free slot to handle this request.
-If this is the case it accepts the session.
-
-When `Session_Accept` is called it will first initialize the server on the platform and then call the servers own `Initialize_Server` procedure.
-The server can then initialize its own backend, e.g. a hard drive controller it drives.
-Once this procedure returns the dispatcher will automatically announce the service on the platform.
-If `Session_Accept` is not called the dispatcher will tell the platform that the request cannot be handled.
-At last the `Session_Cleanup` procedure is called.
-This happens always in the `Dispatch` procedure and will clean all servers where the client has disconnected, similar to a garbage collector.
-
-The `Event` function is called once a request arrives.
-The server interface provides subprograms similar to the client interface to consume and answer these requests but this is omitted here.
-
-The `Block_Size` function is called when the client wants to know how large a single block is.
-It checks first if it is called from the correct instance and returns a static size in this case.
-If it has been called from another instance it will return `0`.
-This instance checking functionality is especially useful if several instances with different properties run simultaneously.
-The same accounts for the `Initialize_Server` procedure which in this case will only print the provided label.
+The usage of the log session here is equivalent to the first example.
+Also the initialization of the timer session is similar.
+The main difference is that the `Timer.Client` package is a generic that needs to be instantiated with a procedure.
+When `Set_Timeout` is called on this instance a timeout is set on the platform.
+The platform will then call the `Event` procedure when this timeout triggers.
+Since the `Event` procedure has no arguments and can be used in multiple contexts no preconditions can be set.
+This requires an initialization check of the timer session.
+Some more specialized callbacks can provide preconditions that provide certain guarantees about initialized arguments.
```

Comment:
What are "more specialized callbacks"? Something the user can implement? Something that Gneiss provides? We should be more specific here.

Reply:
Addressed by https://github.com/Componolit/gneiss/pull/92/commits/db86420560310e1980fe1d0a46a73f93c1080751.